extension/src/background/index.ts

```diff
@@ -33,9 +33,16 @@ async function connectToWebSocket() {
 // Connect to previous roomId using websocket
 async function connectToPreviousRoom() {
     const roomId = await fetchPreviousRoomId();
-    wsClient
-        .getWebSocket()
-        .send(JSON.stringify({ type: MessageType.CHANGE_ROOM, roomId }));
+    sendMessageThroughWebSocket({ type: MessageType.CHANGE_ROOM, roomId });
+}
+
+function sendMessageThroughWebSocket(message: any) {
+    const webSocket = wsClient.getWebSocket();
```

Comment:
Let's make sure the connection is open when we fetch it. We can create it if needed

Reply:
I'm not sure about this for the following reasons:
- I want it to be a pure getter, and not having to return `WebSocket | undefined` and handle the `undefined` case each time I call it
- The connection phase is already handled with the `connect` method, which returns a Promise resolved when the connection is open

Reply:
I have transformed this code a lot in my PR. Maybe it would have made sense to start your PR on top of mine : https://github.com/LesGars/WatchWithMe/pull/86